r/R/arrow-package.R

```diff
@@ -31,6 +31,11 @@
 
 #' @importFrom vctrs s3_register vec_size vec_cast vec_unique
 .onLoad <- function(...) {
+  if (arrow_available()) {
+    # Make sure C++ knows on which thread it is safe to call the R API
+    InitializeMainRThread()
```

Comment:
Do we know for a fact that the R thread never changes?  For example, in JS, there is always "one thread" but the actual thread id can change from iteration to iteration of the event loop.